provision-runners/action.yaml

```diff
@@ -0,0 +1,60 @@
+name: "Provision EC2 Runner"
+description: "Provision a GitHub Action runner in AWS"
+inputs: 
+  aws_access_key_id: 
+    description: 'Use a secret - this account needs "ec2-admin" permissions'
+    required: true
+  aws_secret_access_key: 
+    description: 'Use a secret - this account needs "ec2-admin" permissions'
+    required: true
+  runner_admin_token: 
+    description: "Use a secret. Token used to register a github action runner. If D6E Automaton token make sure it is added to the repo with Admin powers."
+    required: true
+  region: 
+    description: 'aws region for the runner. Default us-east-1'
+    required: false
+    default: 'us-east-1'
+  ami_id: 
+    description: "ami ID that will run on the mac2 instance. Must be associated with the Host Resource Group license."
+    required: true
+  ami_home_directory:
+    description: "the home directory for users on this AMI. Ex: /home"
+    required: true
+    default: /home
+  ami_user: 
+    description: "the automatically configured user on the ami. Usually ec2-user"
+    required: true
+  instance_type: 
+    description: "the instance type to spin up in the Host Resource Group"
+    required: true
+    default: 'mac2.metal'
+  host_resource_group_arn: 
+    description: "the arn of the HRG that will automate setup of the dedicated hosts for the runners"
+    required: false
+    default: ""
+  github_runner_installer:
+    description: "the url of the tar archive for the github runner installer. Must match desired instance architecture"
```

Comment:
I understand now why you said there is friction using this action. 

I think having just one action for all runners is good. 

One way this action could be simplified is by adding a parameter `runner arch` which will have defaults for `ami_id`, `ami_home_directory`, `github_runner_installer`, etc. for each of the supported runners. 

These parameters can still be left as optional in case a 'power user' wants to override these settings. That would make possible something like choosing an Ubuntu ARM64, but change the instance type to something more powerful.

Reply:
Agreed - a lot of this logic could be pushed down a level.